extensions/notebook/src/common/utils.ts

```diff
@@ -30,6 +30,14 @@ export async function mkDir(dirPath: string, outputChannel?: vscode.OutputChanne
 		await fs.ensureDir(dirPath);
 	}
 }
+export function mkDirSync(dirPath: string, outputChannel?: vscode.OutputChannel): void {
```

Comment:
Honestly - if you're looking for speed enhancements I'd get rid of the pathExistsSync check. ensureDirSync will only create the folder if it doesn't exist so checking that it doesn't first is kind of pointless. (you can decide if you care about the output channel message - I personally would just change it to something like "Ensuring folder {0} exists..."

The "check if file exists and if it doesn't create it" is generally considered an anti-pattern anyways. There's nothing preventing the folder from being created between the exists check and when you try to create it. So the suggested method is just to always try to create and then ignore any "already exists" errors.

Reply:
Checking if a file exists in a separate call is even discouraged in the file system module documentation: https://nodejs.org/api/fs.html#fs_fs_stat_path_options_callback

"Using fs.stat() to check for the existence of a file before calling fs.open(), fs.readFile() or fs.writeFile() is not recommended. Instead, user code should open/read/write the file directly and handle the error raised if the file is not available." There are similar blurbs for other file checking methods.

Reply:
Certainly makes a lot of sense - thanks for letting me know! I'll making the changes.